applications/RomApplication/python_scripts/rom_manager.py

```diff
@@ -0,0 +1,526 @@
+        self.rom_training_parameters = self._SetRomTrainintParameters()
+        self.hrom_training_parameters = self.SetHromTrainingParameters()
+        self.CustomizeSimulation = CustomizeSimulation
+        self.UpdateProjectParameters = UpdateProjectParameters
+
+
+
+    def Fit(self, mu_train=None):
+        if mu_train is None:
+            mu_train = ['single case with parameters already contained in the ProjectParameters.json and CustomSimulation']
+        #######################
+        ######  Galerkin ######
+        if self.general_rom_manager_parameters["projection_strategy"].GetString() == "galerkin":
+            training_stages = self.general_rom_manager_parameters["rom_stages_to_train"].GetStringArray()
+            if any(item == "ROM" for item in training_stages):
+                fom_snapshots = self.LaunchTrainROM(mu_train)
+                self._ChangeRomFlags(simulation_to_run = "GalerkinROM")
+                rom_snapshots = self.LaunchROM(mu_train)
+                self.ROMvsFOM_train = np.linalg.norm(fom_snapshots - rom_snapshots)/ np.linalg.norm(fom_snapshots)
+
+            if any(item == "HROM" for item in training_stages):
+                #FIXME there will be an error if we only train HROM, but not ROM
+                self._ChangeRomFlags(simulation_to_run = "trainHROMGalerkin")
+                self.LaunchTrainHROM(mu_train)
+                self._ChangeRomFlags(simulation_to_run = "runHROMGalerkin")
+                hrom_snapshots = self.LaunchHROM(mu_train)
+                self.ROMvsHROM_train = np.linalg.norm(rom_snapshots - hrom_snapshots) / np.linalg.norm(rom_snapshots)
+        #######################
+
+
+        #######################################
+        ##  Least-Squares Petrov Galerkin   ###
+        elif self.general_rom_manager_parameters["projection_strategy"].GetString() == "lspg":
+            training_stages = self.general_rom_manager_parameters["rom_stages_to_train"].GetStringArray()
+            if any(item == "ROM" for item in training_stages):
+                fom_snapshots = self.LaunchTrainROM(mu_train)
+                self._ChangeRomFlags(simulation_to_run = "lspg")
+                rom_snapshots = self.LaunchROM(mu_train)
+                self.ROMvsFOM_train = np.linalg.norm(fom_snapshots - rom_snapshots)/ np.linalg.norm(fom_snapshots)
+            if any(item == "HROM" for item in training_stages):
+                raise Exception('Sorry, Hyper Reduction not yet implemented for lspg')
+        #######################################
+
+
+        ##########################
+        ###  Petrov Galerkin   ###
+        elif self.general_rom_manager_parameters["projection_strategy"].GetString() == "petrov_galerkin":
+            training_stages = self.general_rom_manager_parameters["rom_stages_to_train"].GetStringArray()
+            if any(item == "ROM" for item in training_stages):
+                fom_snapshots = self.LaunchTrainROM(mu_train)
```

Comment:
For a future PR, I suggest making the validation optional or writing the validation snapshots to disks, as we may run out of memory for most simulations.